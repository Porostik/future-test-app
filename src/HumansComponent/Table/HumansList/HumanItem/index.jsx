import React from 'react';
import './HumanItem.scss';
import ItemBlock from './ItemBlock';

// Компонент отображающий данные о человеке
// humanObj - объект с данными о человеке
// onClick - функция нажатия на компонент
function HumanItem({ humanObj, onClick }) {
  return (
    <div className="human-item" onClick={onClick}>
      {Object.values(humanObj).map((item, index) => (
        <ItemBlock key={index} value={item} />
      ))}
    </div>
  );
}

export default HumanItem;
