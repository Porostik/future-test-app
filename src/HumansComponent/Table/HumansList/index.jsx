import React from 'react';
import './HumanList.scss';
import HumanItem from './HumanItem';
import { displayedProps } from 'constants.js';
import { getDisplayedObj } from 'utils';

// Список компонентов "человека"
function HumansList({ items, itemOnClick }) {
  return (
    <div className="human-list">
      {items.map((item, index) => (
        <HumanItem
          key={index}
          humanObj={getDisplayedObj(item, displayedProps)}
          onClick={() => itemOnClick(item)}
        />
      ))}
    </div>
  );
}

export default HumansList;
