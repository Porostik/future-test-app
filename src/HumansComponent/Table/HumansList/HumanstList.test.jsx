import React from 'react';
import { shallow } from 'enzyme';
import HumansList from '.';

describe('Humans list component', () => {
  test('Humans list snapshot', () => {
    const humans = [
      {
        id: 101,
        firstName: 'Sue',
        lastName: 'Corson',
        email: 'DWhalley@in.gov',
        phone: '(612)211-6296',
        address: {
          streetAddress: '9792 Mattis Ct',
          city: 'Waukesha',
          state: 'WI',
          zip: '22178',
        },
        description: 'et lacus magna dolor...',
      },
      {
        id: 102,
        firstName: 'Sue',
        lastName: 'Corson',
        email: 'DWhalley@in.gov',
        phone: '(612)211-6296',
        address: {
          streetAddress: '9792 Mattis Ct',
          city: 'Waukesha',
          state: 'WI',
          zip: '22178',
        },
        description: 'et lacus magna dolor...',
      },
      {
        id: 103,
        firstName: 'Sue',
        lastName: 'Corson',
        email: 'DWhalley@in.gov',
        phone: '(612)211-6296',
        address: {
          streetAddress: '9792 Mattis Ct',
          city: 'Waukesha',
          state: 'WI',
          zip: '22178',
        },
        description: 'et lacus magna dolor...',
      },
    ];
    const wrapper = shallow(<HumansList items={humans} itemOnClick={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });
});
