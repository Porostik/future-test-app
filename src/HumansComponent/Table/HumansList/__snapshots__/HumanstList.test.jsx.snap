// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Humans list component Humans list snapshot 1`] = `
<div
  className="human-list"
>
  <HumanItem
    humanObj={
      Object {
        "email": "DWhalley@in.gov",
        "firstName": "Sue",
        "id": 101,
        "lastName": "Corson",
        "phone": "(612)211-6296",
      }
    }
    key="0"
    onClick={[Function]}
  />
  <HumanItem
    humanObj={
      Object {
        "email": "DWhalley@in.gov",
        "firstName": "Sue",
        "id": 102,
        "lastName": "Corson",
        "phone": "(612)211-6296",
      }
    }
    key="1"
    onClick={[Function]}
  />
  <HumanItem
    humanObj={
      Object {
        "email": "DWhalley@in.gov",
        "firstName": "Sue",
        "id": 103,
        "lastName": "Corson",
        "phone": "(612)211-6296",
      }
    }
    key="2"
    onClick={[Function]}
  />
</div>
`;
